packages/block-editor/src/store/selectors.js

```diff
@@ -1866,21 +1870,22 @@ export function canRemoveBlock( state, clientId ) {
 		return false;
 	}
 
-	const blockEditingMode = getBlockEditingMode( state, rootClientId );
+	const blockEditingMode = getBlockEditingMode( state, clientId );
```

Comment:
It didn't make sense to me that we check the `root` here, so I've changed it to check the clientId being removed.

Reply:
That's true. Maybe I should change it back to see if it causes any issues.

I do agree with the idea of making the logic similar in both selectors, but the issue I'm wondering about is the way `blockEditingMode` works. Usually it's possible to have a `contentOnly` parent block with a `default` (enabled) inner blocks. That's how the Post Content block works when the 'Show Template option is used.

I guess the difference is that after your recent changes, the new logic is that `blockEditingMode: contentOnly` now works differently for container blocks that have the `content` role and it enforces that the children must also have a `content` role and be `contentOnly`. 🤔

Reply:
I've updated the selectors to have matching logic, and it looks like everything still works, so that's good 😄 

I think we could consider moving more of the logic related to insertion, moving, deletion for blocks within sections into a single function, perhaps repurpose `isContainerInsertableToInContentOnlyMode`, it seems like an easy way to keep the code always consistent across the three selectors.

I'll leave it till later on though.